src/server/utils/data_structure.hpp

```diff
@@ -11,19 +11,12 @@
 #include "capio/constants.hpp"
 #include "capio_file.hpp"
 
-struct linux_dirent {
-    unsigned long d_ino;
-    off_t d_off;
-    unsigned short d_reclen;
-    char d_name[DNAME_LENGTH + 2];
-};
-
 struct linux_dirent64 {
-    ino64_t d_ino;
-    off64_t d_off;
-    unsigned short d_reclen;
-    unsigned char d_type;
-    char d_name[DNAME_LENGTH + 1];
+    ino64_t d_ino;           /* 64-bit inode number */
+    off64_t d_off;           /* 64-bit offset to next structure */
+    unsigned short d_reclen; /* Size of this dirent */
+    unsigned char d_type;    /* File type */
+    char d_name[NAME_MAX];   /* Filename (null-terminated) */
```

Comment:
I see you removed the `DNAME_LENGTH` both here and in the `store_dirent` implementation. Is that constant still used somewhere else? If not, you can remove it.

Reply:
corect. removed and use the NAME_MAX constant defined in limits.h which tells the max name length for filenames. MAcro will be removed